### PR TITLE
PWN-5814 - update history list to designs

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5814

## Description of Work

PWN-5814 - update history list to designs

## Screenshot
Android Emulator - Resizable_API_33:5554 2023-02-08 16-37-28

### CODE REVIEW

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/model/HistoryTransaction.kt [hunk 1]
```diff
@@ -90,6 +92,8 @@ sealed class HistoryTransaction(
         val isSend: Boolean
             get() = type == TransferType.SEND
 
+        fun getTokenUrl(): String? = tokenData.iconUrl
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/model/HistoryTransaction.kt:95]: getTokenIconUrl() or even make it a property

File: app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/TransactionViewHolder.kt [hunk 2]
```diff
@@ -34,12 +37,16 @@ class TransactionViewHolder(
 
     private fun showBurnOrMint(transaction: HistoryTransaction.BurnOrMint) {
         with(binding) {
-            transactionTokenImageView.setTransactionIcon(transaction.getIcon())
+            transactionTokenImageView.apply {
+                transaction.getTokenUrl()?.let {
+                    setTokenImage(glideManager, it)
+                } ?: setTransactionIcon(transaction.getIcon())
+            }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/TransactionViewHolder.kt:44]:
```
            transactionTokenImageView.apply {
                transaction.getTokenUrl()
                    ?.also { setTokenImage(glideManager, it) } 
                    ?: setTransactionIcon(transaction.getIcon())
            }
```

File: app/src/main/java/org/p2p/wallet/user/repository/UserInMemoryRepository.kt [hunk 3]
```diff
@@ -2,9 +2,9 @@ package org.p2p.wallet.user.repository
 
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
+import org.p2p.core.token.TokenData
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/user/repository/UserInMemoryRepository.kt:5]: do we have any problem with import ordering? 
if I use reformat code in this class, my imports would be like this:
Screen Shot 2023-02-09 at 13 21 41

File: app/src/main/java/org/p2p/wallet/utils/KeyExtensions.kt [hunk 4]
```diff
@@ -26,3 +26,9 @@ fun String.cutEnd(): String {
     val lastFour = this.takeLast(4)
     return "$firstSixteen...$lastFour"
 }
+
+@Suppress("MagicNumber")
+fun String.cutStart(): String {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/utils/KeyExtensions.kt:31]: please provide int to cut some number of symbols

File: core/src/main/java/org/p2p/core/glide/GlideManager.kt [hunk 5]
```diff
@@ -32,8 +32,12 @@ class GlideManager(context: Context) {
...
                 .with(imageView)
                 .load(url)
-                .placeholder(R.drawable.ic_placeholder_image)
-                .into(imageView)
+                .placeholder(R.drawable.ic_placeholder_image).apply {
+                    val target = if (circleCrop) {
+                        circleCrop()
+                    } else this
+                    target.into(imageView)
+                }
```
Comment (gslevinkov) [inline core/src/main/java/org/p2p/core/glide/GlideManager.kt:40]:
```suggestion
                .placeholder(R.drawable.ic_placeholder_image)
                .run { if (circleCrop) circleCrop() else this }
                .into(imageView)
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/atoms/UiKitTransactionImageView.kt [hunk 6]
```diff
@@ -34,11 +39,20 @@ class UiKitTransactionImageView @JvmOverloads constructor(
         }
     }
 
-    fun setTransactionIcon(@DrawableRes iconRes: Int) {
-        binding.transactionTokenImageView.setImageResource(iconRes)
+    fun setTransactionIcon(@DrawableRes iconRes: Int, iconPadding: Int = defaultIconPadding) = with(binding) {
+        frameToken.setPadding(iconPadding, iconPadding, iconPadding, iconPadding)
+        binding.imageViewToken.setImageResource(iconRes)
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/atoms/UiKitTransactionImageView.kt:44]: no need for binding

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/TransactionViewHolder.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/TransactionViewHolder.kt:44]: each action is a separate line, like in literature

File: app/src/main/java/org/p2p/wallet/user/repository/UserInMemoryRepository.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/user/repository/UserInMemoryRepository.kt:5]: fixed in https://github.com/p2p-org/key-app-android/pull/1220/commits#:~:text=NO%2DTICKET%20%2D%20fix%20import%20ordering%20problems

File: app/src/main/java/org/p2p/wallet/utils/KeyExtensions.kt (hunk 4 above)
Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/utils/KeyExtensions.kt:31]: you can use 4 as Default, and move it to so magic const x D

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/utils/KeyExtensions.kt:31]: kek

**Review by NikolaiEmelianov — APPROVED** [review]

**Pull request merged**